model/trace/http.yaml

```diff
@@ -56,6 +56,9 @@ groups:
     brief: 'Semantic Convention for HTTP Server'
     attributes:
       - ref: http.route
+      - ref: http.request.header
+        sampling_relevant: true
+        requirement_level: opt_in
```

Comment:
The combination of "sampling relevant" together with "opt-in" made me wonder for a bit, as those attributes will likely not be present in most of the cases.

However, I guess although it makes less sense from the point of view of generic samplers, it might be useful in cases where users implement very use-case specific samplers (as I understand was the case in the original issue).

Reply:
good point

what I _want_ to say here is:

* if you write generic instrumentation and provide an option for users to specify headers to be captured, e.g. in Java we have `OTEL_INSTRUMENTATION_HTTP_SERVER_CAPTURE_REQUEST_HEADERS`, then those header attributes are sampling relevant

what I _don't_ want to say here is:

* if you capture these headers as an end-user, e.g. though an instrumentation callback, then those header attributes are sampling relelvant

Reply:
one proposal for addressing this: https://github.com/open-telemetry/semantic-conventions/issues/422#issuecomment-1787471694

Reply:
I believe this is addressed now by #486